### PR TITLE
user registration test

### PR DESCRIPTION
**Pull request by varff**

### CODE REVIEW

**Review by DrGermanius — COMMENTED** [review]
Is it ready for review?

**Comment by varff** [comment]
Nope

File: pkg/handlers/auth_handler.go [hunk 1]
```diff
@@ -13,26 +13,27 @@ import (
 )
 
 type Handler struct {
-	service *services.Service
-	logDb   *logging.LogDb
+	service services.IService
+	logDb   logging.ILogDb
+	resp    models.UserResponse
```
Comment (DrGermanius) [inline pkg/handlers/auth_handler.go:18]: I don't agree with these solution

Comment (DrGermanius) [reply pkg/handlers/auth_handler.go:18]: Didn't get the point of this

Comment (DrGermanius) [reply pkg/handlers/auth_handler.go:18]: If you need to use different sctures you will add all of them? I think it's not necessary

File: pkg/handlers/service_handler_test.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+		return
+	}
+	c.Request, err = http.NewRequest(http.MethodPost, "", strings.NewReader(string(jPhone)))
+	ctl := gomock.NewController(t)
+	mockIService := mock.NewMockIService(ctl)
+	mockLogDb := mock.NewMockILogDb(ctl)
+	mockResp := mock.NewMockUserResponse(ctl)
+	hand := NewHandler(mockIService, mockLogDb, mockResp)
+	mockIService.EXPECT().LogoutUser(gomock.Any(), gomock.Any()).Return(nil)
+	mockIService.EXPECT().DeleteUser(gomock.Any()).Return(apperrors.ErrNotFound)
```
Comment (DrGermanius) [inline pkg/handlers/service_handler_test.go:139]: Are you sure about using any()? It must be only in corner-cases

File: pkg/models/response.go [hunk 3]
```diff
@@ -6,6 +6,13 @@ import (
 	"github.com/varff/InnoUserService/pkg/logging"
 )
 
+type Responser struct{}
+type UserResponse interface {
+	NewOkResponse(message string) *OkResponse
+	NewTokenResponse(token string) *TokenResponse
+	NewErrorResponse(logDb *logging.LogDb, c *gin.Context, err string) *ErrorResponse
+}
```
Comment (DrGermanius) [inline pkg/models/response.go:14]: nonononono

Comment (DrGermanius) [reply pkg/models/response.go:14]: let's talk about it later

**Review by DrGermanius — CHANGES_REQUESTED** [review]

File: pkg/handlers/service_handler_test.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+	var c *gin.Context
+	w := httptest.NewRecorder()
+
+	c, _ = gin.CreateTestContext(w)
+	c.AddParam("expires", "10")
+	phone.Phone = "+375292222222"
+	phoneInt, err := strconv.Atoi(phone.Phone)
+	if err != nil {
+		return
+	}
```
Comment (DrGermanius) [inline pkg/handlers/service_handler_test.go:36]: need to use t.Fatal or something

**Review by DrGermanius — CHANGES_REQUESTED** [review]
Please fix and merge

**Pull request merged**